.github/workflows/unit.yaml

```diff
@@ -35,3 +35,46 @@ jobs:
       if: ${{ !((github.event_name == 'repository_dispatch' && github.event.client_payload.slash_command.args.named.sha != '' && contains(github.event.client_payload.pull_request.head.sha, github.event.client_payload.slash_command.args.named.sha)) || inputs.skipRefCheck) }}
       with:
         script: core.setFailed('Ref is not latest')
+  go-cover:
```

Comment:
I may be missing something but is there a reason why you made go-cover a separate job? couldn't you add the same steps to the previous job?

Reply:
i like separated too. Is there a reason this is under the unit workflow instead of a separate workflow entirely?

I remember gocover requiring unit tests to generate coverage at some point so that might be a reason but I haven't looked recently.

Reply:
but if gocover requires unit tests first I don't understand how this works since it's a separate job

Reply:
Actually I see how this is working. It's taking the local coverage.out instead of regenerating. Same problem as here https://github.com/Azure/aks-app-routing-operator/pull/74/files#r1277680016

Reply:
https://github.com/Azure/aks-app-routing-operator/pull/74#discussion_r1277711868